### PR TITLE
Add 'Ephemera box' field to folder edit form, fixes #1143

### LINKED ISSUE #1143: Ephemera: Allow moving boxless folders into boxes
This is to support recording the boxes for the ~4K posters migrated into Figgy as boxless.  Important to make this as efficient as possible, given the volume of manual work.

Ideal workflow:
1. Search on poster title
2. Edit poster
3. Select box on edit form

### PR DESCRIPTION
**Pull request by hackmastera**
screen shot 2018-05-09 at 1 59 48 pm

### CODE REVIEW

File: app/controllers/ephemera_folders_controller.rb [hunk 1]
```diff
@@ -88,13 +92,40 @@ def load_fields
...
+      []
+    end.unshift(nil_box)
+  end
+
+  def ephemera_box
+    parent_resource.is_a?(EphemeraBox) ? parent_resource : nil
+  end
+
+  def nil_box
+    Struct.new(:title, :id).new('', '')
```
Comment (tpendragon) [inline app/controllers/ephemera_folders_controller.rb]: What's the nil_box get us that the empty array doesn't?

Comment (hackmastera) [reply app/controllers/ephemera_folders_controller.rb]: I actually don't think I tested the empty array case. When you have a non-empty array you need a nil option to keep the form from filling in the first value of the array, and allowing the user to not fill in the field if they don't want to.

File: app/controllers/ephemera_folders_controller.rb [hunk 2]
```diff
@@ -88,13 +92,40 @@ def load_fields
     end
   end
 
+  # provides instance variables needed to populate collection and selected for append_id
+  # @see app/views/ephemera_folders/edit_fields/_append_id.html.erb
+  def load_boxes
+    instance_variable_set("@available_boxes", available_boxes)
```
[the remark this answers is not in the document]
Comment (hackmastera) [reply app/controllers/ephemera_folders_controller.rb]: I must have seen it floating around somewhere else and gotten it stuck in my head; I will change.

[the remark this answers is not in the document]
Comment (hackmastera) [reply app/controllers/ephemera_folders_controller.rb]: oh, in the method above.

File: .rubocop.yml [hunk 3]
```diff
@@ -69,6 +69,7 @@ Metrics/ClassLength:
     - 'app/services/manifest_builder/manifest_service_locator.rb'
     - 'app/services/pdf_generator/cover_page_generator.rb'
     - 'app/decorators/valkyrie/resource_decorator.rb'
+    - 'app/controllers/ephemera_folders_controller.rb'
```
Comment (escowles) [inline .rubocop.yml]: For next week: maybe we should increase the class length? Or look at these classes and see if there's any refactoring we want to do?

**Pull request merged**